build.zig

```diff
@@ -1,7 +1,38 @@
 const std = @import("std");
 
 pub fn build(b: *std.Build) void {
+    const target = b.standardTargetOptions(.{});
+    const optimize = b.standardOptimizeOption(.{});
+
     _ = b.addModule("diffz", .{
         .root_source_file = b.path("DiffMatchPatch.zig"),
+        .target = target,
+        .optimize = optimize,
+    });
+
+    // Run tests
+    const tests = b.addTest(.{
+        .name = "tests",
+        .root_source_file = b.path("DiffMatchPatch.zig"),
+        .target = target,
+        .optimize = optimize,
+    });
+    const step_tests = b.addRunArtifact(tests);
+
+    b.step("test", "Run diffz tests").dependOn(&step_tests.step);
+
+    // Adds a step to generate code coverage
+    const cov_step = b.step("cov", "Generate coverage (kcov must be installed)");
+
+    const cov_run = b.addSystemCommand(&.{
+        "kcov",
+        "--clean",
+        "--include-pattern=DiffMatchPatch.zig",
+        "--exclude-line=unreachable,expect(false)",
+        "kcov-output",
     });
+    cov_run.addArtifactArg(tests);
+    cov_step.dependOn(&cov_run.step);
+    _ = cov_run.captureStdOut();
+    _ = cov_run.captureStdErr();
```

Comment:
The Zig Build System provides functions to declare input and output parameters that will properly deal with caching and dependencies between build steps.
This also avoids the addition of `kcov-output` in the `.gitignore`. The coverage will be stored in `zig-out/coverage`.

I currently can't test this code so there may be some error in it. Shouldn't be a problem since I know it worked for a different project I work on https://github.com/zigtools/zig-lsp-codegen/blob/478d5833cfbc7701bd369a0dd0a53c86b52cbd14/build.zig#L84

```suggestion
    const addOutputDirectoryArg = comptime if (@import("builtin").zig_version.order(.{ .major = 0, .minor = 13, .patch = 0 }) == .lt)
        std.Build.Step.Run.addOutputFileArg
    else
        std.Build.Step.Run.addOutputDirectoryArg;

    const run_kcov = b.addSystemCommand(&.{
        "kcov",
        "--clean",
        "--exclude-line=unreachable,expect(false)",
    });
    run_kcov.addPrefixedDirectoryArg("--include-pattern=", b.path("."));
    const coverage_output = addOutputDirectoryArg(run_kcov, ".");
    run_kcov.addArtifactArg(tests);
    run_kcov.enableTestRunnerMode();

    const install_coverage = b.addInstallDirectory(.{
        .source_dir = coverage_output,
        .install_dir = .{ .custom = "coverage" },
        .install_subdir = "",
    });

    const coverage_step = b.step("coverage", "Generate coverage (kcov must be installed)");
    coverage_step.dependOn(&install_coverage.step);
```

The `addOutputDirectoryArg` has a check to not break compatibility with Zig 0.12.0

Reply:
I'll give it a try and let you know. 

Actually while I'm thinking about it, could you bake this into a patch? That way we don't have to back out the change if something happens. Thanks.

It would be nice if there were a "copy this to clipboard as patch" button but GitHub has never liked patches much.

Reply:
Does GitHub not offer you to commit this patch? https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

Reply:
https://github.com/mnemnion/diffz/pull/2

Reply:
I'll just go ahead and do that, sure. I was going to make a side branch and check that it works out there, but there's not a lot of advantage in doing so.